be/tool/beck.c

```diff
@@ -73,10 +73,29 @@
 M0_TL_DESCR_DECLARE(ad_domains, M0_EXTERN);
 M0_TL_DECLARE(ad_domains, M0_EXTERN, struct ad_domain_map);
 
-struct queue;
+struct queue {
+	pthread_mutex_t q_lock;
```

Comment:
Why don't we use m0_mutex?

Reply:
@huanghua78 : I think you are right we should have done these and other changes you mentioned below and we will do these maybe as a part of beck enhancement since we will need to test these a lot. Since this code needs to land today for the code-freeze we will file a new Jira for these changes in the meanwhile you can approve these changes for now.